password/password.cabal

```diff
@@ -200,23 +203,28 @@ test-suite password-tasty
   default-language:
       Haskell2010
 
-  if os(darwin)
-    cpp-options: -DIS_MAC_OS
-  else
-    build-depends: scrypt
-
   if flag(argon2)
     cpp-options:
       -DCABAL_FLAG_argon2
+
   if flag(bcrypt)
     cpp-options:
       -DCABAL_FLAG_bcrypt
+
   if flag(pbkdf2)
     cpp-options:
       -DCABAL_FLAG_pbkdf2
+
   if flag(scrypt)
     cpp-options:
       -DCABAL_FLAG_scrypt
+
+    -- The scrypt package is optionally used in tests to ensure compatibility
+    -- between password and scrypt.  However, scrypt doesn't work on OSX, so
+    -- we only add the scrypt dependency if we're not building on OSX.
+    if !os(darwin)
+      build-depends: scrypt
```

Comment:
With this logic, we only add the `scrypt` Haskell package build-depend if BOTH the following are true:

- The `scrypt` cabal flag is turned ON

and

- We are NOT building on darwin

I think this captures the logic we are expecting for whether or not to be able to run the scrypt tests.